process/coordinator/process.go

```diff
@@ -952,39 +952,38 @@ func (tc *transactionCoordinator) CreateReceiptsHash() ([]byte, error) {
 	return finalReceiptHash, err
 }
 
-// CreateMarshalizedReceipts will return all the receipts hashes list in one marshalized object
+// CreateMarshalizedReceipts will return all the receipts list in one marshalized object
 func (tc *transactionCoordinator) CreateMarshalizedReceipts() ([]byte, error) {
-	receiptsHashes := make([][]byte, 0)
+	receiptsBatch := &batch.Batch{}
 	for _, blockType := range tc.keysInterimProcs {
 		interProc, ok := tc.interimProcessors[blockType]
 		if !ok {
 			continue
 		}
 
-		mb := interProc.GetCreatedInShardMiniBlock()
-		if mb == nil {
+		miniBlock := interProc.GetCreatedInShardMiniBlock()
+		if miniBlock == nil {
 			continue
 		}
 
-		mbHash, err := core.CalculateHash(tc.marshalizer, tc.hasher, mb)
+		marshalizedMiniBlock, err := tc.marshalizer.Marshal(miniBlock)
```

Comment:
this is a list of marshalled miniblocks. make sure that these things are not saved 2 times.

Reply:
They are not saved anywhere as these are the ones which are deleted from applyBodyToHeader through newBody := deleteSelfReceiptsMiniBlocks(body) method